src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    public static final int INITIAL_SIZE = 10;
+
+    private T[] array = (T[]) new Object[INITIAL_SIZE];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        checking();
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        generateIndexException(index < 0);
+        checking();
+        if (index == size) {
+            add(value);
+        } else {
+            T t = array[index];
+            generateIndexException(t == null);
+            System.arraycopy(array, index, array, index + 1, size - index);
+            array[index] = value;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        generateIndexException(index >= size || index < 0);
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        generateIndexException(index >= size || index < 0);
```

Comment:
Again, use a single method to check the index validity to adhere to the DRY principle.

Reply:
Done